### PR TITLE
Add a test for the fast-path pack()/unpack() functions.

### PR DESCRIPTION
**Pull request by bangerth**
This is a follow-up to #5800.

### CODE REVIEW

**Comment by tjhei** [comment]
Looks good. /run-tests

File: tests/base/utilities_pack_unpack_03.cc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+
+template <typename T>
+void check (const T &object)
+{
+  const std::vector<char> buffer = Utilities::pack (object);
+  if (!(buffer.size() == sizeof(object))
+      ||
+      !(std::memcmp(buffer.data(), &object, buffer.size()) == 0)
```
Comment (masterleinad) [inline tests/base/utilities_pack_unpack_03.cc]: Can you elaborate on this check? Doesn't this mean that you expect `buffer.data()` to return a `nullptr`?

Comment (bangerth) [reply tests/base/utilities_pack_unpack_03.cc]: No -- `std::memcmp` compares the data pointed to by the first two arguments (both pointers) with length given by the last argument and returns zero if the two memory locations have identical content.

Comment (masterleinad) [reply tests/base/utilities_pack_unpack_03.cc]: Ah, I see. I confused it with `std::memcpy`.

Comment (tjhei) [reply tests/base/utilities_pack_unpack_03.cc]: I had to stare at this if clause for quite a while to understand it if that makes you feel better. ;-)

Comment (tjhei) [reply tests/base/utilities_pack_unpack_03.cc]: I would have written them as separate AssertThrows.

**Pull request merged**